nflow-engine/src/main/java/io/nflow/engine/service/WorkflowDefinitionService.java

```diff
@@ -71,49 +52,45 @@ public void setWorkflowDefinitions(@NFlow AbstractResource nflowNonSpringWorkflo
 
   /**
    * Return all managed workflow definitions.
+   *
    * @return List of workflow definitions.
    */
   public List<AbstractWorkflowDefinition<? extends WorkflowState>> getWorkflowDefinitions() {
-    return new ArrayList<>(workflowDefinitions.values());
+    synchronized (workflowDefinitions) {
```

Comment:
Is synchronized needed here? workflowDefinitions is already a synchronizedMap.

Reply:
Yes it is, "It is imperative that the user manually synchronize on the returned map when iterating over any of its collection views"